src/Aspire.Hosting/ContainerResourceBuilderExtensions.cs

```diff
@@ -297,6 +306,14 @@ public static IResourceBuilder<T> WithImage<T>(this IResourceBuilder<T> builder,
             imageAnnotation.Tag = parsedReference.Tag ?? tag ?? "latest";
         }
 
+        // If there's a DockerfileBuildAnnotation with an image name/tag, clear them
+        // so that the user's explicit image preference is respected
+        if (builder.Resource.Annotations.OfType<DockerfileBuildAnnotation>().SingleOrDefault() is { } buildAnnotation)
```

Comment:
Not sure about this, I think we want to set this as well no? If there's a `DockerfileBuildAnnotation` then we want to set the image (and tag)?